lit_parrot/model.py

```diff
@@ -221,18 +241,28 @@ def forward(
             cache_k, cache_v = cache_k.to(dtype=k.dtype), cache_v.to(dtype=v.dtype)
             # check if reached token limit
             if input_pos[-1] >= max_seq_length:
-                input_pos = torch.tensor(max_seq_length - 1, device=input_pos.device)
+                input_pos = torch.tensor([max_seq_length - 1], device=input_pos.device)
                 # shift 1 position to the left
                 cache_k = torch.roll(cache_k, -1, dims=2)
                 cache_v = torch.roll(cache_v, -1, dims=2)
+            padding_idx = cache_k.size(2) - 1  # send padding data to a padding index, doesn't matter which
+            input_pos = torch.cat(
+                (input_pos, torch.full((padding,), padding_idx, device=input_pos.device, dtype=input_pos.dtype))
+            )
             k = cache_k.index_copy(2, input_pos, k)
             v = cache_v.index_copy(2, input_pos, v)
             kv_cache = k, v
 
-        # efficient attention using Flash Attention CUDA kernels
-        y = F.scaled_dot_product_attention(
-            q, k, v, attn_mask=mask, dropout_p=0.0, scale=1.0 / math.sqrt(self.config.head_size)
-        )
+        scale = 1.0 / math.sqrt(self.config.head_size)
+        if padding == 0:
+            y = F.scaled_dot_product_attention(q, k, v, mask, dropout_p=0.0, scale=scale)
+        else:
+            # `scaled_dot_product_attention` produces nans when padding is added
+            # TODO: create repro and report to pytorch
```

Comment:
Done: https://github.com/pytorch/pytorch/issues/103749

```suggestion
            # `scaled_dot_product_attention` produces nans when padding is added
            # https://github.com/pytorch/pytorch/issues/103749
```